server/aws-lsp-codewhisperer/src/shared/localProjectContextController.ts

```diff
@@ -433,89 +412,42 @@ export class LocalProjectContextController {
-                    .then(async (sourceFiles: string[]) => {
-                        if (!respectUserGitIgnores) {
-                            return sourceFiles
-                        }
-
-                        const userGitIgnoreFilterByFile = new Map(
-                            await Promise.all(
-                                localGitIgnoreFiles.map(async filePath => {
-                                    const filter = ignore()
-                                    try {
-                                        filter.add((await fs.promises.readFile(filePath)).toString())
-                                    } catch (error) {
-                                        this.log.error(`Error reading .gitignore file ${filePath}: ${error}`)
-                                    }
-                                    return [filePath, filter] as const
-                                })
-                            )
-                        )
-
-                        return sourceFiles.reduce((filteredSourceFiles, filePath) => {
-                            if (sizeConstraints.remainingIndexSize <= 0) {
-                                return filteredSourceFiles
-                            }
-
-                            const isIgnored = [...userGitIgnoreFilterByFile].some(
-                                ([gitIgnorePath, filter]: [string, any]) => {
-                                    const gitIgnoreDir = path.dirname(path.resolve(gitIgnorePath))
-                                    const relativePath = path.relative(gitIgnoreDir, filePath)
-
-                                    return !relativePath.startsWith('..') && filter.ignores(relativePath)
-                                }
-                            )
-
-                            if (!isIgnored && this.fileMeetsFileSizeConstraints(filePath, sizeConstraints)) {
-                                filteredSourceFiles.push(filePath)
+        const uniqueFilesToIndex = new Set<string>()
+        let filesExceedingMaxSize = 0
+        for (const folder of workspaceFolders) {
+            const absoluteFolderPath = path.resolve(URI.parse(folder.uri).fsPath)
+            const filesUnderFolder = await listFilesWithGitignore(absoluteFolderPath)
+            for (const file of filesUnderFolder) {
+                const fileExtName = '.' + getFileExtensionName(file)
+                if (!uniqueFilesToIndex.has(file) && fileExtensions?.includes(fileExtName)) {
+                    try {
+                        const fileSize = fs.statSync(file).size
+                        if (fileSize < sizeConstraints.maxFileSize) {
+                            if (sizeConstraints.remainingIndexSize > fileSize) {
+                                uniqueFilesToIndex.add(file)
+                                sizeConstraints.remainingIndexSize = sizeConstraints.remainingIndexSize - fileSize
+                            } else {
+                                this.log.info(
```

Comment:
nit: should be warn for the log?

Reply:
not a warn because this is expected. we restrict on max size.